package NYT.NYTree.NProto;

////////////////////////////////////////////////////////////////////////////////

message TAttributeFilter
{
    required int32 mode = 1;
    repeated string keys = 2;
}

message TReqGet
{
    optional TAttributeFilter attribute_filter = 1;
}

message TRspGet
{
    required bytes value = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqSet
{
    required bytes value = 1;
}

message TRspSet
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqRemove
{ }

message TRspRemove
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqList
{
    repeated string attributes = 1;
}

message TRspList
{
    required bytes keys = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqExists
{ }

message TRspExists
{
    required bool value = 1;
}

////////////////////////////////////////////////////////////////////////////////

